sphinx/themes/basic/static/searchtools.js

```diff
@@ -410,6 +410,7 @@ const Search = {
     // console.info("search results:", Search.lastresults);
 
     // print the results
+    const [_, searchTerms, _excluded, highlightTerms, _objects] = searchParameters;
```

Comment:
```suggestion
    const [_query, searchTerms, _excludedTerms, highlightTerms, _objectTerms] = searchParameters;
```

Reply:
In fact: perhaps refactor-out the creation of the `searchParameters` variable?  (it is no longer much convenience if we need to extract all the items within it anyway)

Reply:
The diff I arrive at is:

```diff
diff --git a/sphinx/themes/basic/static/searchtools.js b/sphinx/themes/basic/static/searchtools.js
index 3295a48b3..92da3f8b2 100644
--- a/sphinx/themes/basic/static/searchtools.js
+++ b/sphinx/themes/basic/static/searchtools.js
@@ -402,15 +402,14 @@ const Search = {
   },
 
   query: (query) => {
-    const searchParameters = Search._parseQuery(query);
-    const results = Search._performSearch(...searchParameters);
+    const [searchQuery, searchTerms, excludedTerms, highlightTerms, objectTerms] = Search._parseQuery(query);
+    const results = Search._performSearch(searchQuery, searchTerms, excludedTerms, highlightTerms, objectTerms);
 
     // for debugging
     //Search.lastresults = results.slice();  // a copy
     // console.info("search results:", Search.lastresults);
 
     // print the results
-    const [_, searchTerms, _excluded, highlightTerms, _objects] = searchParameters;
     _displayNextItem(results, results.length, searchTerms, highlightTerms);
   },
 

```

(not sure I can offer that as a suggestion using the GitHub UI though, sorry)

Reply:
By looking at the file I think we need to clean it up (my IDE was complaining about many things and I think we can make the private interface cleaner, but it'll be a separate PR).